pyaedt/edb_core/hfss.py

```diff
@@ -78,23 +78,279 @@ def get_trace_width_for_traces_with_ports(self):
 
         Returns
         -------
-        list
-           List of names of the created ports.
+        str
+            Port Name.
 
         """
-        pinList = convert_py_list_to_net_list(pin_list)
-        if self._hfss_terminals.CreateHfssPorts(self._builder, pinList):
-            return True
+        return self.parent.core_siwave.create_circuit_port_on_pin(pos_pin, neg_pin, impedance, port_name)
+
+    @aedt_exception_handler
+    def create_voltage_source_on_pin(self, pos_pin, neg_pin, voltage_value=3.3, phase_value=0, source_name=""):
+        """Create a voltage source.
+
+        Parameters
+        ----------
+        pos_pin : Object
+            Positive Pin.
+        neg_pin : Object
+            Negative Pin.
+        voltage_value : float, optional
+            Value for the voltage. The default is ``3.3``.
+        phase_value : optional
+            Value for the phase. The default is ``0``.
+        source_name : str, optional
+            Name of the source. The default is ``""``.
+
+        Returns
+        -------
+        str
+            Source Name
+
+        Examples
+        --------
+
+        >>> from pyaedt import Edb
+        >>> edbapp = Edb("myaedbfolder", "project name", "release version")
+        >>> pins =edbapp.core_components.get_pin_from_component("U2A5")
+        >>> edbapp.core_hfss.create_voltage_source_on_pin(pins[0], pins[1],50,"source_name")
+        """
+        return self.parent.core_siwave.create_voltage_source_on_pin(pos_pin, neg_pin, voltage_value, phase_value, source_name)
+
+    @aedt_exception_handler
+    def create_current_source_on_pin(self, pos_pin, neg_pin, current_value=0.1, phase_value=0, source_name=""):
+        """Create a current source.
+
+        Parameters
+        ----------
+        pos_pin : Object
+            Positive Pin.
+        neg_pin : Object
+            Negative Pin.
+        current_value : float, optional
+            Value for the current. The default is ``0.1``.
+        phase_value: optional
+            Value for the phase. The default is ``0``.
+        source_name : str, optional
+            Name of the source. The default is ``""``.
+
+        Returns
+        -------
+        str
+            Source Name.
+
+        Examples
+        --------
+
+        >>> from pyaedt import Edb
+        >>> edbapp = Edb("myaedbfolder", "project name", "release version")
+        >>> pins =edbapp.core_components.get_pin_from_component("U2A5")
+        >>> edbapp.core_hfss.create_current_source_on_pin(pins[0], pins[1],50,"source_name")
+        """
+
+        return self.parent.core_siwave.create_current_source_on_pin(pos_pin, neg_pin, current_value, phase_value, source_name)
+
+    @aedt_exception_handler
+    def create_resistor_on_pin(self, pos_pin, neg_pin, rvalue=1, resistor_name=""):
+        """Create a voltage source.
+
+        Parameters
+        ----------
+        pos_pin : Object
```

Comment:
Can we get more specific with the type?  If we can provide an exact type, users will get a type link walk through the docs.

Reply:
@akaszynski it is an API specific internal object type. I could enter the name but will be useless for the documentation

Reply:
Ah, ok.  Thanks!